tensorflow_serving/core/servable_state_monitor.cc

```diff
@@ -210,7 +211,11 @@ bool ServableStateMonitor::WaitUntilServablesReachState(
         reached_goal_state = incoming_reached_goal_state;
         notified.Notify();
       });
-  notified.WaitForNotification();
+  if (timeout > 0) {
+    WaitForNotificationWithTimeout(&notified, timeout_in_us);
```

Comment:
Given that you are merely forwarding to an existing utility method, and that testing this would be nontrivial, I don't think a test case is needed.